Comments in configuration

steff-o opened the pull request:
Closes #1380 

Support for c-style one line comments: `// This is a comment` and multi line comments: `/* This is also a comment */`. It does NOT support nested multiline comments and does not fix any syntactical errors that may occur due to parts being commented out. All characters, except new lines, in comments are replaced with spaces in order to keep same size and line numbering.

Also added better json parsing error message which throws an error containing the row number and an excerpt of the configuration at the location of the error instead of just file position.

tonnyandersson commented:
Unable to use multi-line (`/* ... */`) comment on strings containing slashes, for example:

`/* "url": "https://github.com/origo-map/origo" */`

or

```
/*
  {
    "name": "link",
    "options": {
      "title": "Origo",
      "url": "https://github.com/origo-map/origo"
    }
  },
*/
```

tonnyandersson commented:
Also, please use one of the [supported linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link the PR to the corresponding issue.

"Fix" is in the list, but I think the "for" breaks the spell.

steff-o commented:
Ooops. I made a terrible logical error after refactoring the code and did not test thoroughly after. Fixed that and also fixed that the last character is removed if the comment reaches end of input.

tonnyandersson commented:
LGTM.
